lib/src/parsers/utilities/parser-utility.ts

```diff
@@ -173,6 +186,83 @@ export function extractNumberProperty(
   return { value, location, line };
 }
 
+/**
+ * Extract an optional object literal property value from an object literal.
+ *
+ * @param objectLiteral an object literal
+ * @param propertyName the property to extract
+ */
+export function extractOptionalObjectProperty(
+  objectLiteral: ObjectLiteralExpression,
+  propertyName: string
+): Locatable<ObjectLiteralExpression> | undefined {
+  const property = objectLiteral.getProperty(propertyName);
+  if (property) {
+    if (!TypeGuards.isPropertyAssignment(property)) {
+      throw new Error("expected property assignment");
+    }
+    const value = property.getInitializerIfKindOrThrow(
+      ts.SyntaxKind.ObjectLiteralExpression
+    );
+    const location = property.getSourceFile().getFilePath();
+    const line = value.getStartLineNumber();
+
+    return { value, location, line };
+  } else {
+    return;
+  }
+}
+
+/**
+ * Extract an optional array literal property value from an object literal.
+ *
+ * @param objectLiteral an object literal
+ * @param propertyName the property to extract
+ */
+export function extractOptionalArrayProperty(
+  objectLiteral: ObjectLiteralExpression,
+  propertyName: string
+): Locatable<ArrayLiteralExpression> | undefined {
+  const property = objectLiteral.getProperty(propertyName);
+  if (property) {
+    if (!TypeGuards.isPropertyAssignment(property)) {
+      throw new Error("expected property assignment");
+    }
+    const value = property.getInitializerIfKindOrThrow(
+      ts.SyntaxKind.ArrayLiteralExpression
+    );
+    const location = property.getSourceFile().getFilePath();
+    const line = value.getStartLineNumber();
+
+    return { value, location, line };
+  } else {
+    return;
```

Comment:
Maybe `return undefined` for clarity? You could also just remove this entirely (`return undefined` is the default behaviour).